### PR TITLE
chore: add cluster id to header

### PR DESCRIPTION
**Pull request by mscb402**

### CODE REVIEW

**Review by tokers — COMMENTED** [review]
1. Use `http.Header` to instead `map[string]string` in `httpClient` methods.
2. Set the storage interface to `httpClient` to avoid a lot of changes

File: api.go [hunk 1]
```diff
@@ -174,10 +174,12 @@ func newAPI(cli httpClient) APIInterface {
...
 func (impl *apiImpl) CreateAPI(ctx context.Context, api *API, opts *ResourceCreateOptions) (*API, error) {
 	var createdAPI API
-
+	if !ensureClusterID(impl.client, opts) {
+		return nil, ErrClusterIDNotExist
+	}
 	appID := opts.Application.ID
 	uri := path.Join(_apiPathPrefix, "apps", appID.String(), "apis")
-	err := impl.client.sendPostRequest(ctx, uri, "", api, jsonPayloadDecodeFactory(&createdAPI))
+	err := impl.client.sendPostRequest(ctx, uri, "", api, jsonPayloadDecodeFactory(&createdAPI), appendHeader(mapClusterIdFromHttpClient(impl.client), mapClusterIdFromOpts(opts)))
```
Comment (tokers) [inline api.go:182]: We don't need this: `mapClusterIdFromHttpClient(impl.client)`. How to choose the Cluster ID should be done by the `httpClient`. Use the one in storage or, overriding from the given one (from opts).

Comment (mscb402) [reply api.go]: Removed

File: examples/sanity/go.mod [hunk 2]
```diff
@@ -2,11 +2,12 @@ module sanity
 
 go 1.19
 
-replace github.com/api7/cloud-go-sdk => /Users/alex/Workstation/tokers/api7/cloud-go-sdk
+replace github.com/api7/cloud-go-sdk => /Users/chenbeng/Documents/api7/cloud-go-sdk
```
Comment (nic-6443) [inline examples/sanity/go.mod:5]: Use relative path?

Comment (mscb402) [reply examples/sanity/go.mod:5]: this is just for test

**Review by nic-6443 — APPROVED** [review]

**Review by tokers — APPROVED** [review]

**Pull request merged**